pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -825,3 +835,34 @@ func newCloud(sess *session.Session, l cloudprovider.Logger) (*Cloud, error) {
 	}
 	return c, nil
 }
+
+// validateMachineType checks whether the given machine type is valid and returns true if so
+func validateMachineType(machine string) bool {
+	// These are currently hardcoded as AWS does not expose them via the API
+	machinePrefixes := []string{"t2", "m3", "m4"}
+	machineSuffixes := []string{"nano", "micro", "small", "medium", "large", "xlarge", "2xlarge"}
+	machineSplit := strings.Split(machine, ".")
+	if len(machineSplit) != 2 {
+		return false
+	}
+	prefixCorrect := false
+	suffixCorrect := false
+
+	for _, p := range machinePrefixes {
+		if p == machineSplit[0] {
```

Comment:
Will that not return false if the any one of them doesn't match? or does range work differently to how I think it does